devicetypes/smartthings/zigbee-ceiling-fan-light.src/led-fan-lightings.groovy

```diff
@@ -0,0 +1,181 @@
+			}
+			tileAttribute("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action: "switch level.setLevel"
+			}
+		}
+
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: "", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		
+		main(["switch"])
+		details(["switch", "refresh", "switchLevel"])
+	}
+}
+
+private getFAN_CLUSTER_VALUE() { 0x0202 }
+private getFAN_STATUS_VALUE() { 0x0000 }
+private getON_OFF_CLUSTER_VALUE() { 0x0006 }
+
+def parse(String description) {
+	// Parse incoming device messages to generate events
+	log.debug "description is $description"
+	def event = zigbee.getEvent(description)	
+	if (event) {
+		sendEvent(event)
+	} else if (description?.startsWith('read attr -')) {
+		def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+		if (zigbeeMap.clusterInt == FAN_CLUSTER_VALUE &&
+		    zigbeeMap.attrInt == FAN_STATUS_VALUE) {		
+			log.debug "read attribute event for fan cluster attrib FAN_STATUS"
+			def childDevice = childDevices.find {
+				//find light child device
+				log.debug "parse() child device found"
+				it.device.deviceNetworkId == "${device.deviceNetworkId}:1" 
+			}
+			def fanSpeedEvent = createEvent(name: "fanSpeed", value: zigbeeMap.value as Integer)
+			childDevice.sendEvent(fanSpeedEvent)
+			if (fanSpeedEvent.value == 0) {
+				log.debug "fan_off => switch off"
+				childDevice.sendEvent(name: "switch", value: "off")
+				childDevice.sendEvent(name: "level", value: 0)	// For cloud to cloud device UI update
+			} else {
+				log.debug "fan_on => switch on"
+				childDevice.sendEvent(name: "switch", value: "on")
+				def int_v = fanSpeedEvent.value
+				int_v = int_v * 25
+				int_v = int_v > 100 ? 100 : int_v
+				log.debug "child device Level set $int_v"
+				childDevice.sendEvent(name: "level", value: int_v)	// For cloud to cloud device UI update
+			}
+		}		
+	} else {
+		def cluster = zigbee.parse(description)
+		if (cluster && 
+		    cluster.clusterInt == ON_OFF_CLUSTER_VALUE &&
+		    cluster.command == 0x07) {
+			if (cluster.data[0] == 0x00) {
+				log.debug "ON/OFF REPORTING CONFIG RESPONSE: " + cluster
+				sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+			} else {
+				log.warn "ON/OFF REPORTING CONFIG FAILED- error code:${cluster.data[0]}"
+			}
+		}
+	}
+}
+
+def off() {
+	zigbee.off()
+}
+
+def on() {
+	zigbee.on()
+}
+
+def setLevel(value) {
+	zigbee.setLevel(value)
+}
+
+private sendFanSpeed(val) {
+	delayBetween([zigbee.writeAttribute(FAN_CLUSTER_VALUE, FAN_STATUS_VALUE, DataType.ENUM8, val), zigbee.readAttribute(FAN_CLUSTER_VALUE, FAN_STATUS_VALUE)], 100)
+}
+
+def ping() {
+	// PING is used by Device-Watch in attempt to reach the Device
+	return zigbee.onOffRefresh()
+}
```

Comment:
As Steven has said, you also need this method to return a command that would cause sending an event for child device, e.g. `zigbee.readAttribute(FAN_CLUSTER_VALUE, FAN_STATUS_VALUE)`

Reply:
OK. I see.